### PR TITLE
Add support for existing networks to quilt module

### PR DESCRIPTION
**Pull request by akarve**
Things I considered but decided not to do:
1. `existing_network` as `map(str)` but this is problematic beacause validation conditions can only reference _themselves_ leading to external conditions and more complicated validation logic
2. Make `cidr` nullable (so existing_network users don't have to set it) but this is problematic for the locals in `vpc` (and would complicate that code) so even existing_network users need to pass `cidr` and we at least do size validation on it.
3. Consolidating `existing_network_requires` and `new_network_requires` — these are (kinda) logical duals of one another but attempting to consolidate conditions hurt readability.

### CODE REVIEW

**Review by dimaryaz — CHANGES_REQUESTED** [review]

File: modules/quilt/variables.tf [hunk 1]
```diff
@@ -14,18 +14,17 @@ variable "cidr" {
...
   type        = bool
   nullable    = false
-  description = "Whether CloudFormation template uses internet-facing or internal ELBs"
+  description = "If true create an inward ELBv2, else create an internet-facing ELBv2."
 }
 
 variable "db_snapshot_identifier" {
   type        = string
-  nullable    = true
   default     = null
```
Comment (dimaryaz) [inline modules/quilt/variables.tf:28]: Should the default now be `""`?

File: modules/quilt/variables.tf [hunk 2]
```diff
@@ -119,3 +118,46 @@ variable "parameters" {
...
+  default     = null
+  description = "VPC endpoint for API Gateway (api-execute) for Quilt services."
+
+}
+
+variable "intra_subnets" {
+  type        = list(string)
+  default     = null
+  description = "Intra subnets need not have Internet access as they only communicate with private subnets."
+  validation {
```
Comment (dimaryaz) [inline modules/quilt/variables.tf:139]: If Cidr validation is in the VPC module, then maybe subnets should be, too?

File: modules/quilt/variables.tf (hunk 1 above)
Comment (akarve) [reply modules/quilt/variables.tf:28]: In theory `nullable=true` is the default and this change has no effect. In practice, I will revert this change.

File: modules/quilt/variables.tf (hunk 2 above)
Comment (akarve) [reply modules/quilt/variables.tf:139]: Sure that makes sense.

File: modules/quilt/variables.tf [hunk 3]
```diff
@@ -119,3 +119,34 @@ variable "parameters" {
...
+variable "private_subnets" {
+  type        = list(string)
+  default     = null
+  description = "Have Internet access to reach public AWS services."
+}
+
+variable "public_subnets" {
+  type        = list(string)
+  default     = null
+  description = "Only needed when var.internal == False (for NAT & load balancer)."
```
Comment (dimaryaz) [inline modules/quilt/variables.tf:151]: Lowercase false?

**Review by dimaryaz — APPROVED** [review]

File: modules/quilt/variables.tf (hunk 3 above)
Comment (akarve) [reply modules/quilt/variables.tf]: Fixed

**Pull request merged**